processor/pom.xml

```diff
@@ -92,6 +92,17 @@
         <plugins>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestEntries>
+                            <MapStruct-Generated-Comment>${project.groupId}:${project.artifactId}:${project.version}</MapStruct-Generated-Comment>
```

Comment:
That's what I meant with fiddling. I have some test code in place to examine where the APT-implementation comes from, and it would probably burn down to a combination of system-properties and jar file name (e.g. in case of the Eclipse implementation, I want to know which one as well)... Still need to figure out how to condense the information and not create too large strings. I already did some tests with JDKs 1.6 .. 1.9 and the Eclipse compiler bundled by felix, but not yet with the plain-vanilla Eclipse...

But anyway, Build-JDK does not help - as you found out yourself just now... :wink:

I would suggest you finish this PR and leave the issue open, I'll just add my stuff along side with it and create a new PR then.